Fix code signing in Xcode 8

neonichu opened the pull request:
This should fix building Pods for device using Xcode 8 beta.

Fixes #5523

DanToml commented:
LGTM - Glad that it somewhat keeps old behaviour when `ProvisioningStyle` is missing.

DanToml commented:
When you rebase this, it'll need to go in `Installer::Xcode::PodsProjectGenerator#prepare`

orta commented on lib/cocoapods/installer.rb:
```diff
@@ -594,6 +594,8 @@ def prepare_pods_project
           build_configuration.build_settings['STRIP_INSTALLED_PRODUCT'] = 'NO'
           build_configuration.build_settings['CLANG_ENABLE_OBJC_ARC'] = 'YES'
+          build_configuration.build_settings['CODE_SIGNING_REQUIRED'] = 'NO'
+          build_configuration.build_settings['PROVISIONING_PROFILE_SPECIFIER'] = 'AAAAAAAAAA/'
```
Open up and say `AAAAAAA\`

![Giphy](http://media3.giphy.com/media/JavtlJ2FXJbnq/giphy.gif)

mrackwitz replied:
I think we should use a self-explanatory value here, which can referred to (e.g. `NO_SIGNING/` if underscores would be acceptable).

neonichu replied:
or `YAATUSABES`

segiddins commented:
👍 with a rebase, test coverage, and a changelog

mrackwitz commented:
:+1: beside my comment…

opswhisperer commented:
This is not working for me

![screenshot](https://www.dropbox.com/s/xu0s3kdheyaxatv/Screenshot%202016-06-21%2007.57.43.jpg?dl=0)

neonichu commented:
@opswhisperer looks as if you're only using this branch, you also need CocoaPods/Xcodeproj#387 — also the error in Xcode's General tab will not go away — we can only fix building at the moment.

opswhisperer commented:
Ok, missed that. Even with that I'm getting a different error now than before
`No signing certificate 'iOS Development' found:  No 'iOS Development' signing certificate with a private key was found.`
For each one of my targets

opswhisperer commented:
Never mind all good!

warpling commented:
Is Travis stuck?

neonichu commented:
I'm not quite sure about the result differences in the validator unit specs and also the integration specs:

```diff
-          LD_RUNPATH_SEARCH_PATHS: $(inherited) @executable_path/Frameworks @loader_path/Frameworks
+          LD_RUNPATH_SEARCH_PATHS:
+            - $(inherited)
+            - '@executable_path/Frameworks'
+            - '@loader_path/Frameworks'
```

```diff
-          ARCHS: $(ARCHS_STANDARD_64_BIT)
-          CODE_SIGN_IDENTITY[sdk=iphoneos*]: iPhone Developer
+          ARCHS:
+            - $(ARCHS_STANDARD_64_BIT)
+          CODE_SIGN_IDENTITY[sdk=iphoneos*]: ''
```

Seems like some array-y things become single strings and vice versa when using Xcode 8 — would appreciate a second set of eyes to determine if this is something to be addressed in Xcodeproj or not.

segiddins commented on lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:
```diff
@@ -45,12 +45,13 @@ def target_definition
+              'PODS_ROOT'                         => '$(SRCROOT)',
+              'PRODUCT_BUNDLE_IDENTIFIER'         => 'org.cocoapods.${PRODUCT_NAME:rfc1034identifier}',
+              'SKIP_INSTALL'                      => 'YES',
+              'CODE_SIGN_IDENTITY[sdk=iphoneos*]' => '',
```
please keep the list sorted

neonichu replied:
Ah right, didn't notice that the list was sorted :)

segiddins commented:
@neonichu if it still works in Xcode 7 as an array, then I wouldn't worry about it

mrackwitz commented:
>@neonichu if it still works in Xcode 7 as an array, then I wouldn't worry about it

If the single values are managed together as strings as seen from the changes in the unit tests, this could potentially lead to issues when relying on e.g. `uniq` to avoid duplicate values. What has caused this change?

neonichu commented:
@mrackwitz Using Xcode 8 instead of 7.3.1 to run the tests — no changes to CP except the ones in this PR when running them.

neonichu commented:
I reverted the unit test changes, since we're still running those against 7.3 on CI and updated the integration specs in CocoaPods/cocoapods-integration-specs#74

This should be good to go now 👍

neonichu commented:
Travis 👷 finished now 👌

tsheaff commented:
When i this getting merged to master?

opswhisperer commented:
We also need CocoaPods/Xcodeproj#387 which seems to have been deleted.

neonichu commented:
@opswhisperer that shouldn't be needed anymore, the settings where transferred to CP itself in order to not mess with usage of `xcodeproj` outside of the CP context.

opswhisperer commented:
Odd, because I have 1.0.1 and I'm still having code signing errors. It looks as if the settings are correct but I get build errors. I forced a reinstall of xcodeproj because I had the deleted PR installed locally.

```
Signing for AFNetworking requires a development team. Select a development team in the Target Editor.
Code signing is required for product type 'Framework' in SDK 'iOS 10.0'
```

neonichu commented:
This isn't included in 1.0.1 — it has just been merged to master so far

opswhisperer commented:
Ahh that makes sense, thanks

aurawindsurfing commented:
Hi, can you point to me where I find info how to install those not relesed changes? Should I simply pull this repo? How will that work with existing install? I'm sorry for asking, just had trouble finding the info on the obvious. Cheers!

orta commented:
There is a guide on this: https://guides.cocoapods.org/using/a-gemfile.html

tal-sh commented:
My project contains multiple targets, and different signing identities. Is something like that could be supported by xcode?

pbernery commented:
Don't know if I'll get an answer now :) : why did you set the `PROVISIONING_PROFILE_SPECIFIER` to `NO_SIGNING/`? 
Is this expected to be changed by the developer with its own prov. profile?

DanToml commented:
@pbernery We sign the frameworks in the embed frameworks script

pbernery commented:
Thanks @DanToml for this quick response.

I am trying to find the right solution to a signing issue related to this. I have the following error when building on the command line (used on my CI):
```
=== BUILD TARGET Alamofire OF PROJECT Pods WITH CONFIGURATION Release ===

Check dependencies
Alamofire does not support provisioning profiles. Alamofire does not support provisioning profiles, but provisioning profile NO_SIGNING/ has been manually specified. Set the provisioning profile value to "Automatic" in the build settings editor.
```
I got this error for all Pods. Removing "NO_SIGNING/" fixes this issue and lets Xcode continue.

I did not find any issue in the CocoaPods GitHub related to "NO_SIGNING/", and am trying to understand the build/sign workflow expected by CP. 

The dependency resolution of Xcode makes the Pods project targets compile first. It seems that Xcode 8.3.2 checks provisioning profiles as soon as something is set in this field, even though "Don't code sign" is specified in these projects. Is this a changed behaviour since the inclusion of this patch or is there something wrong in my project?

I made a `pod deintegrate && pod install` to make sure I did not changed the Pods project mistakenly.

Daniate commented:
@pbernery I got this error too.
